core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -246,57 +249,67 @@ export class ContributorStatsComponent {
 
     if (response.question_contribution_stats.length > 0) {
       this.statsData.questionContribution = this
-        .createQuestionContributionStatsPage(
-          response.question_contribution_stats);
+        .createAllQuestionStatsRecords(
+          response.question_contribution_stats, true);
     }
 
     if (response.question_review_stats.length > 0) {
-      this.statsData.questionReview = this.createQuestionReviewStatsPage(
-        response.question_review_stats);
+      this.statsData.questionReview = this.createAllQuestionStatsRecords(
+        response.question_review_stats, false);
     }
   }
 
-  createTranslationContributionStatsPage(
-      stat: TranslationContributionBackendDict): StatsPage {
-    return {
-      data: [this.createTranslationContributionStat(stat)],
-      language: this.languageUtilService.getAudioLanguageDescription(
-        stat.language_code),
-      currentPageStartIndex: 0,
-      currentPageEndIndex: 5
-    };
-  }
+  createInitialTranslationStatsRecord(
```

Comment:
This function is messy. I suggest you make two separate functions, one for each case, and get rid of the boolean variable.